src/realm/mixed.cpp

```diff
@@ -428,6 +425,32 @@ Decimal128 Mixed::export_to_type() const noexcept
     return {};
 }
 
+template <>
+StringData Mixed::export_to_type() const noexcept
+{
+    REALM_ASSERT(m_type);
```

Comment:
This is general, but also minor stuff. You don't need to use `if/else if`, since most of the time you are just returning a value.